Libraries/Image/RCTImageLoader.m

```diff
@@ -610,31 +611,26 @@ - (RCTImageLoaderCancellationBlock)decodeImageDataWithoutClipping:(NSData *)data
   }
 }
 
-- (RCTImageLoaderCancellationBlock)getImageSize:(NSString *)imageTag
-                                          block:(void(^)(NSError *error, CGSize size))completionBlock
-{
-  return [self loadImageOrDataWithTag:imageTag
-                                 size:CGSizeZero
-                                scale:1
-                           resizeMode:RCTResizeModeStretch
-                        progressBlock:nil
-                      completionBlock:^(NSError *error, id imageOrData) {
-                        CGSize size;
-                        if ([imageOrData isKindOfClass:[NSData class]]) {
-                          NSDictionary *meta = RCTGetImageMetadata(imageOrData);
-                          size = (CGSize){
-                            [meta[(id)kCGImagePropertyPixelWidth] doubleValue],
-                            [meta[(id)kCGImagePropertyPixelHeight] doubleValue],
-                          };
-                        } else {
-                          UIImage *image = imageOrData;
-                          size = (CGSize){
-                            image.size.width * image.scale,
-                            image.size.height * image.scale,
-                          };
-                        }
-                        completionBlock(error, size);
-                      }];
+- (NSDictionary<NSString *, id>*)getMetadataForImage:(id)imageOrData {
```

Comment:
By convention, Objective-C methods that return a value directly should not start with "get". This should just be:

    - (NSDictionary<NSString *, NSNumber *> *)metadataForImage:(id)imageOrData

I'm also not sure why the name was changed however, since the method still seems to just return the image size. If the plan is to add additional metadata in future, this probably won't work anyway because you need the NSData to get most metadata, you can't get it directly from the UIImage.